scikeras/wrappers.py

```diff
@@ -314,14 +335,36 @@ def _validate_data(self, X, y=None, reset=True):
         out : {ndarray, sparse matrix} or tuple of these
             The validated input. A tuple is returned if `y` is not None.
         """
+        # default to TFs backend float type
+        # instead of always float64 (sklearns default)
+        tf_backend_dtype = np.dtype(tf.keras.backend.floatx())
+
+        def get_dtype(arr) -> np.dtype:
```

Comment:
I think there's a typo in this function: it looks like a `np.ndarray` with `arr.dtype.kind == "O"` will needlessly create another array. Maybe this function instead?

``` python
def get_dtype(arr):
    if isinstance(arr, np.ndarray):
        if arr.dtype.kind != "O":
            return arry.dtype
        else:
            return output_dtype

    # arr is not an ndarray
    arr_dtype = np.asarray(arr).dtype
    if arr_dtype.kind != "O":
        return arr_dtype
    return output_dtype
```

Reply:
No, that's on purpose. This is for when we get an iterable of numpy arrays, for example a list of numpy arrays. I tried doing `arr[0].dtype` but that would then fail some tests from the Scikit-Learn checks that specifically check that you do not try to index the inputs before converting them to an array.

Reply:
I misread your comment. You're right. There is a needless conversion. Let me make another PR to fix that.